crypto/signing/mcl/suiteBLS12_381.go

```diff
@@ -41,12 +41,13 @@ func init() {
 
 	pubKey := &bls.PublicKey{}
 	bls.BlsGetGeneratorForPublicKey(pubKey)
+	blsSwapG := bls.IsSwapG()
 	if blsSwapG {
-		generatorG1 := bls.CastG1FromPublicKey(pubKey)
+		generatorG1 := (*bls.G1)(unsafe.Pointer(pubKey))
 		basePointG1Str.Store(generatorG1.GetString(10))
 		basePointG2Str.Store(g2str)
 	} else {
-		generatorG2 := bls.CastG2FromPublicKey(pubKey)
+		generatorG2 := (*bls.G2)(unsafe.Pointer(pubKey))
```

Comment:
please use the casting function provided

Reply:
cast calls bls.IsSwapG(). Added TODO for a later fix